securedrop_client/storage.py

```diff
@@ -544,21 +540,26 @@ def delete_source_collection(journalist_filename: str, data_dir: str) -> None:
 def delete_single_submission_or_reply_on_disk(obj_db: Union[File, Message, Reply],
                                               data_dir: str) -> None:
     """
-    Delete on disk a single submission or reply.
+    Delete on disk any files associated with a single submission or reply.
     """
-    files_to_delete = []
-    filename_without_extensions = obj_db.filename.split('.')[0]
-    file_glob_pattern = os.path.join(
-        data_dir,
-        '{}*'.format(filename_without_extensions)
-    )
-    files_to_delete.extend(glob.glob(file_glob_pattern))
 
-    for file_to_delete in files_to_delete:
+    try:
+        os.remove(obj_db.location(data_dir))
+    except FileNotFoundError:
+        logging.info('Object %s already deleted, skipping', obj_db.location(data_dir))
+
+    if isinstance(obj_db, File):
+        # When a file is being processed (e.g. its filename is being extracted), it
+        # is stored in an intermediate state in a folder in the source data dir.
+        # We should also delete this entire directory.
+        in_progress_folder = os.path.join(
```

Comment:
The name `in_progress_folder` seems inaccurate, as it exists when the `File` has been fully downloaded and decrypted.

Reply:
And it might be more future-proof to use `os.path.dirname(obj_db.location(data_dir))` here, in case we change the paths in `db.py`.